experiments/compositions/expanders/helm-expander/main.go

```diff
@@ -0,0 +1,401 @@
+	// https://helm.sh/docs/helm/helm_lint/
+	// Usage:
+	//  helm lint PATH [flags]
+	//
+	args := []string{"lint", e.path}
+	if e.facade != nil {
+		facadeFile := filepath.Join(e.path, "facade-values.yaml")
+		args = append(args, "-f", facadeFile)
+	}
+
+	op, err := exec.Command("helm", args...).CombinedOutput()
+	if err != nil {
+		result.Error.Message = fmt.Sprintf("failed validating template:\n %s", string(op))
+		result.Status = pb.Status_VALIDATE_FAILED
+		log.Print(result.Error.Message)
+		return result, nil
+	}
+
+	//log.Printf("Result:\n  type: %s\n  error: %s\n  status: %s\n", result.Type, result.Errors, result.Status)
+	return result, nil
+}
+
+func (e *Expander) Evaluate(result *pb.EvaluateResult) (*pb.EvaluateResult, error) {
+	// https://helm.sh/docs/helm/helm_template/
+	// Usage:
+	//  helm template [NAME] [CHART] [flags]
+	//
+	name := "unknown"
+	args := []string{"template", name, e.path}
+	if e.facade != nil {
+		name = e.facade.GetName()
+		facadeFile := filepath.Join(e.path, "facade-values.yaml")
+		args = append(args, "-f", facadeFile)
+	}
+
+	//args = append(args, "--output-dir", filepath.Join(e.path, "rendered"))
+
+	log.Printf("running: helm %s", strings.Join(args, " "))
+	op, err := exec.Command("helm", args...).CombinedOutput()
+	if err != nil {
+		result.Error.Message = fmt.Sprintf("failed evaluating helm chart:\n %s", string(op))
+		result.Status = pb.Status_EVALUATE_FAILED
+		log.Print(result.Error.Message)
+		return result, nil
+	}
+
+	manifests := op
+	/*
+		// If using output-dir we need to collect the generated files
+		//args = append(args, "--output-dir", filepath.Join(e.path, "rendered"))
+			manifests := []byte{}
+			// parse output and read files
+			lines := strings.Split(string(op), "\n")
+			for _, line := range lines {
+				log.Print(line)
+				if strings.HasPrefix(line, "wrote ") {
+					file := strings.TrimPrefix(line, "wrote ")
+					// read file
+					fileContent, err := os.ReadFile(file)
+					if err != nil {
+						newerr := fmt.Errorf("failed reading expanded file: %s, %w", file, err)
+						log.Print(newerr)
+						return nil, newerr
+					}
+					manifests = append(manifests, fileContent...)
+				}
+			}
+	*/
+	result.Manifests = manifests
+
+	//log.Printf("Result:\n  type: %s\n  error: %s\n  status: %s\n  manifests: %s\n  values: %s\n", result.Type, result.Errors, result.Status, result.Manifests, result.Values)
+	return result, nil
+}
+
+// ------------- GRPC Server implementation ----------------
+
+// server is used to implement exander.Evaluator
+type server struct {
+	pb.UnimplementedExpanderServer
+}
+
+// Verify the expander config/template
+func (s *server) Validate(ctx context.Context, req *pb.ValidateRequest) (*pb.ValidateResult, error) {
+	result := &pb.ValidateResult{
+		Status: pb.Status_SUCCESS,
+		Error:  &pb.Error{},
+	}
+
+	log.Printf("Validate called")
+	e, failedMessage, err := NewExpander(ctx, nil, req)
```

Comment:
Might be clearer to have a method Validate on the Expander returned by NewExpander, and have it return pb.ValidateResult